frontend/src/components/Songs/SongInfo.js

```diff
@@ -17,22 +17,19 @@ const SongInfo = () => {
   const dispatch = useDispatch();
   const history = useHistory();
   const { id } = useParams();
-  const [isPlaying, setIsPlaying] = useState(false);
   const song = useSelector((state) => state.songs[id]);
   const artist = useSelector((state) => state.songs[id]?.Artist);
   const album = useSelector((state) => state.songs[id]?.Album);
   const audioFile = useSelector((state) => state.songs[id]?.url);
+  const playerSong = useSelector((state) => state.audio.song);
+  const playerStatus = useSelector((state) => state.audio.status);
+  const isPlaying = song.id === playerSong.id && playerStatus === 'play';
```

Comment:
Instead of using a separate state to keep track of whether the song is playing or not, we can just determine it using the new status field and the song field in the audio store. If the song in the audio store matches the song info id, and the status of the audio store is playing, then we know that the current song info is playing.